dlc-wallet/src/main/scala/org/bitcoins/dlc/wallet/util/DLCStatusBuilder.scala

```diff
@@ -0,0 +1,245 @@
+package org.bitcoins.dlc.wallet.util
+
+import org.bitcoins.core.dlc.accounting.DLCAccounting
+import org.bitcoins.core.protocol.dlc.models.{
+  ClosedDLCStatus,
+  ContractInfo,
+  DLCState,
+  DLCStatus,
+  OracleOutcome
+}
+import org.bitcoins.core.protocol.dlc.models.DLCStatus.{
+  Accepted,
+  Broadcasted,
+  Claimed,
+  Confirmed,
+  Offered,
+  Refunded,
+  RemoteClaimed,
+  Signed
+}
+import org.bitcoins.core.protocol.transaction.Transaction
+import org.bitcoins.crypto.SchnorrDigitalSignature
+import org.bitcoins.dlc.wallet.accounting.AccountingUtil
+import org.bitcoins.dlc.wallet.models.{
+  DLCAcceptDb,
+  DLCContractDataDb,
+  DLCDb,
+  DLCOfferDb,
+  OracleNonceDb
+}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+object DLCStatusBuilder {
+
+  /** Helper method to convert a bunch of indepdendent datastructures into a in progress dlc status */
+  def buildInProgressDLCStatus(
+      state: DLCState.InProgressState,
```

Comment:
why even take `state` as a param here and just use `dlcDb.state` everywhere

Reply:
it is useful because it is a more precise type. The problem with `DlcDb` is that it represents too many states. We need to eventually make sub types to represent the precise state of the DLC

Reply:
Maybe the best solution would be to make a new type (maybe `DLCGlobalData`?) and have it by typed based on the state. Then we can just make a simple `toDLCGlobalData` function on `DLCDb`.